updating docs including section on contribution, CHANGELOG, and some …

shahzebsiddiqui opened the pull request:
…output examples from doc examples

Signed-off-by: Shahzeb Siddiqui <shahzebmsiddiqui@gmail.com>

shahzebsiddiqui committed:
updating docs including section on contribution, CHANGELOG, and some output examples from doc examples

Signed-off-by: Shahzeb Siddiqui <shahzebmsiddiqui@gmail.com>

vsoch commented on docs/contributing.rst:
```diff
@@ -15,123 +13,25 @@ There are many ways you can help contribute to buildtest that may include
+- Shahzeb Siddiqui (**@shahzebsiddiqui**)
 
-- **Stale**  - buildtest is using `Stale <https://github.com/marketplace/stale>`_ to close outdated issues. This is configured in ``.github/stale.yml``. If there is no activity on a issue after certain time period, **probot-stale** will mark the issue and project maintainers can close it manually. For more details on Stale refer to the `documentation <https://probot.github.io/>`_
+- Vanessa Sochat (**@vsoch**)
```
:D

vsoch commented on docs/contributing/github_integration.rst:
```diff
@@ -0,0 +1,115 @@
+GitHub Integration
+====================
+
+buildtest has several github integration, including automated checks during PR that maintainers will check to
```
nit: github integration**s**

vsoch replied:
"including automated checks during PR that maintainers will check during the PR remove" (delete the "to" after check)

vsoch commented on docs/contributing/github_integration.rst:
```diff
@@ -0,0 +1,115 @@
+
+buildtest has several github integration, including automated checks during PR that maintainers will check to
+during the PR review. You should check results from the `buildtest actions <https://github.com/HPC-buildtest/buildtest-framework/actions>`_
+to see if any fixes need to me made.
```
maybe we should add that it's linked from the PR too, something like:

> You should check results from the `buildtest actions <https://github.com/HPC-buildtest/buildtest-framework/actions>`_ that are also typically linked as part of the pull request testing suite.

vsoch commented on docs/contributing/github_integration.rst:
```diff
@@ -0,0 +1,115 @@
+to run regression test and Codecov and Coveralls depend on Travis to pass all checks.
+
+You will want to make sure code is formatted via black as we have automated checks for python formatting. If you have not
+setup the black hook check out :ref:`black_hook`
```
Maybe for now we can add something like:

> If you run the black linter and find that the check still fails, check the step of the action that installs black and ensure that you are using the same version locally.

Another thing we can do is to pin a particular black version and then tell the user up front to use it, also spit out a warning in the commit hook if it's a different version.

shahzebsiddiqui replied:
If we want to pin a version, it may make sense to add this to `docs/requirements.txt` so that all deps are added in one place and user dont have to do this manually. Looking at https://buildtest.readthedocs.io/en/devel/contributing.html#github-hooks i can see the `pip install black==19.3b0` be something user don't have to worry about if its part of `requirements.txt`. Anyhow I will add some section about your comment in the docs.

vsoch replied:
Yeah this is a good idea. Then we will need to install that version explicitly for the test as well!

vsoch commented on docs/contributing/github_integration.rst:
```diff
@@ -0,0 +1,115 @@
+- **Travis**: https://travis-ci.com/HPC-buildtest/buildtest-framework
+
+- **CodeCov**: https://codecov.io/gh/HPC-buildtest/buildtest-framework
+
```
Would it make sense to put the links to the apps alongside the descriptions so it doesn't look like they are listed twice / redundant?

shahzebsiddiqui replied:
I agree, i will fix this

vsoch commented on docs/contributing/github_integration.rst:
```diff
@@ -0,0 +1,115 @@
+GitHub Bots
+-----------
+
+Buildtest has a few bots to do various operations that are described below.
```
One thing to bring up for general discussion - I've seen buildtest with upper and lower case B, is the convention to always use lowercase unless we are starting a sentence?

shahzebsiddiqui replied:
Good point, i guess i can change it to lower case to be consistent even if grammatically it may make no sense. The name is `buildtest` so it may be confusing with `Buildtest`

vsoch replied:
Great! Yeah as long as we are consistent we shouldn't see derivations popping up like `Buildtest` or even `BuildTest`.

vsoch commented on docs/contributing/release_process.rst:
```diff
@@ -1,6 +1,15 @@
+
+Any incoming Pull Request should be assigned to one or more maintainers for review. Upon approval, the PR should
+be **Squash and Merge**. If it's important to preserve a few commits during PR then **Rebase and merge** is acceptable.
+
```
Maybe we should add a note about cleaning up the final squashed message if appropriate?

vsoch commented on docs/contributing/setup.rst:
```diff
@@ -90,8 +85,60 @@ Once you are ready to push to your fork repo do the following::
+    git push origin <feature-branch>
+
+
+General Tips
```
How do these tips relate to the CONTRIBUTING.md? Or a better question - what goes here vs. there?

shahzebsiddiqui replied:
good point i dont know the best place, let me try putting this in another location.

vsoch commented on docs/contributing/setup.rst:
```diff
@@ -90,8 +85,60 @@ Once you are ready to push to your fork repo do the following::
 
-Someone from the **buildtest team** will review the PR and get back to you with the feedback. If the reviewer requests
-some changes, then the user is requested to make changes and update the branch used for sending PR
+5. Please don't open a duplicate issue if their is an existing issue addressing similar problem, instead you can participate in discussion in the issue or contact appropriate individuals directly in slack
```
if _there_ is an existing issue

vsoch reviewed (changes requested):
Overall looks good! Just a few nits and questions.

shahzebsiddiqui committed:
second round of changes to documentation addressing comments from @vsoch

rename release_process.rst --> maintainer_guide.rst and add few more
sections dedicated to maintainers.

shahzebsiddiqui committed:
rsync with devel

vsoch commented on docs/contributing/github_integration.rst:
```diff
@@ -1,47 +1,35 @@
 
-- **CodeFactor**: https://www.codefactor.io/repository/github/hpc-buildtest/buildtest-framework
+- `Coveralls <https://coveralls.io/github/HPC-buildtest/buildtest-framework>`_ - Coveralls is a web service to help you track your code coverage over time, and ensure that all your new code is fully covered.
 
```
This is much cleaner! One question that comes to mind is that (for some future update) if we can store the repository address / name as a variable, so we don't need to do manual changes for links like this. I'm not sure the extent to which this is possible with sphinx, but maybe? Either way, that's out of scope for here, this looks good.

vsoch commented on docs/contributing/maintainer_guide.rst:
```diff
@@ -0,0 +1,75 @@
+
+1. Any incoming Pull Request should be assigned to one or more maintainers for review.
+
+2. Upon approval, the PR should be **Squash and Merge**. If it's important to preserve a few commits during PR then **Rebase and Merge** is acceptable.
```
I think there is a setting for this under the repo settings, I'll check after this final review.
